code/go/pkg/linkedfiles/fs.go

```diff
@@ -0,0 +1,47 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package linkedfiles
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+)
+
+var _ fs.FS = (*LinksFS)(nil)
+
+// LinksFS is a filesystem that handles linked files.
+// It wraps another filesystem and checks for linked files with the ".link" extension.
+// If a linked file is found, it reads the link file to determine the target file
+// and its checksum. If the target file is up to date, it returns the target file.
+// Otherwise, it returns an error.
+type LinksFS struct {
```

Comment:
We are going to publicly expose this FS, but not sure if this should be used out of the spec validation.

Do you think this can be useful in elastic-package? What do you think about moving it to an internal package by now?

In any case I think we can leave the `Link` type in a public package as it can be more clearly reused in elastic-package.

I would be fine also with letting everything here in a private package, and have its own implementation in elastic-package, as the needings in both places are not exactly the same.

Reply:
I think makes sense to move it to a private package and let ep handle it on its own